### PR TITLE
Rename register() for container services

### PR DESCRIPTION
**Pull request by othercorey**
Suggest not using a core function name for optional feature.

If php had proper function overloading, then we could just do it that way.

### CODE REVIEW

**Comment by ndm2** [comment]
This feels a little bit like an outlier, given the other method names in the class/interface. What about just `services`, or are there maybe plans to introduce `registerRoutes`, `registerCommands`, etc?

**Comment by ADmad** [comment]
I too was thinking if we could just go with `services()`.

**Comment by othercorey** [comment]
> I too was thinking if we could just go with `services()`.

That would fit.

File: src/Http/BaseApplication.php [hunk 1]
```diff
@@ -249,22 +249,22 @@ public function getContainer(): ContainerInterface
         if ($this->container) {
             return $this->container;
         }
-        $container = $this->register(new Container());
+        $container = $this->services(new Container());
         foreach ($this->plugins->with('register') as $plugin) {
```
Comment (ndm2) [inline src/Http/BaseApplication.php:253]: I guess this should then be `services` too, with the `BasePlugin::$registerEnabled` property being renamed to `servicesEnabled` accordingly.

Comment (othercorey) [reply src/Http/BaseApplication.php:253]: thats true

**Comment by markstory** [comment]
I'm ok with the method being `services`. I used the name `register` as that is the name league uses in ServiceProviders. However, `services` is more aligned with the other method names in Application.

**Review by ADmad — APPROVED** [review]

**Pull request merged**